### ISSUE TITLE
Incorrect error message: Module 'torch' has no 'device' member (no-member)

### ISSUE THREAD

**Issue by joaqo** [microsoft/pyright#1746]
Hi, I am getting this incorrect error message when using pytorch, here are the steps to reproduce:

`pip install torch`

```python
import torch

torch.device("cuda")  # This line causes the error message
```

Its weird because I get the `Module 'torch' has no 'device' member (no-member)` error, but using `go to definition` works perfectly fine when done on `device`.

These are my pyright settings:

```json
 {
    "python.formatting.provider": "black",
    "python.linting.pylintEnabled": true,
    "python.linting.enabled": true,
    "python.pythonPath": "python",
    "python.analysis.typeCheckingMode": "off",
    "python.analysis.typeCheckingMode": "off",
    "python.formatting.blackArgs": ["--line-length=98"]
}
```

Thanks!

**Comment by erictraut** [comment]
Which version of torch do you have installed? There were typing issues with older versions of torch, and I think this was one of the symptoms. I just verified that with the latest version, this works fine. In any case, this is not an issue with pyright but with the type information in the torch package.

**Comment by joaqo** [comment]
Hey, thanks for the reply! I dug a little deeper, and the issue is a bit strange.

If I make a venv with just a torch==1.8.1 install I get the following pip freeze, and things work just fine, just like you said.
```bash
numpy==1.20.2
torch==1.8.1
typing-extensions==3.7.4.3
```

Now, if I install a local library, using poetry, I get the following pip freeze, and the error pops up back again.
```bash
appdirs @ file:///home/lalo/.cache/pypoetry/artifacts/47/cf/4f/4ef02fb715aa36daeebad18cc5570126159c659c41c7b5ec46a7387d9b/appdirs-1.4.4-py2.py3-none-any.whl
astroid @ file:///home/lalo/.cache/pypoetry/artifacts/34/e0/99/2edbd67ade70c8c00b3e1ab5559e9ed5efc5470f97256888886f6d312a/astroid-2.4.2-py3-none-any.whl
attrs @ file:///home/lalo/.cache/pypoetry/artifacts/f9/48/82/553e4bef24d3b294c0c18f27a7853f3ed151508efd144cb7ea37db1c48/attrs-20.3.0-py2.py3-none-any.whl
black @ file:///home/lalo/.cache/pypoetry/artifacts/56/00/d0/a3bee2cbd161f599bac7d8ee24fb2b17aeba39b53bc42724de21f352bf/black-20.8b1.tar.gz
click @ file:///home/lalo/.cache/pypoetry/artifacts/e2/79/34/a23e9d2f683ed66be11ec3bd760dec3a2fe228cfdedf2071bcf0531b06/click-7.1.2-py2.py3-none-any.whl
colorama @ file:///home/lalo/.cache/pypoetry/artifacts/9e/b3/11/7d87ac44fdb2d557301f1f4086a37c080d1482a98751abe7cdbabbad26/colorama-0.4.4-py2.py3-none-any.whl
commonmark @ file:///home/lalo/.cache/pypoetry/artifacts/11/56/f6/d054064b623fab5c7e4420f60d931f49fea2dacdebe1dc991201010c84/commonmark-0.9.1-py2.py3-none-any.whl
cycler @ file:///home/lalo/.cache/pypoetry/artifacts/af/e8/be/e7e2a18c4317324d4e3beeec6f4366506194f59bad1c2fbe91a54445eb/cycler-0.10.0-py2.py3-none-any.whl
distlib @ file:///home/lalo/.cache/pypoetry/artifacts/0d/93/5f/4559a04b5d9c0e0d7d9369a0b09fd059a38e96525f6bf97ee0c851710f/distlib-0.3.1-py2.py3-none-any.whl
filelock @ file:///home/lalo/.cache/pypoetry/artifacts/7e/c4/97/2cfbeab3cc292d0b4290cb7cab0b969b3002dc24f6dd5944cbe340e684/filelock-3.0.12-py3-none-any.whl
filterpy==1.4.5
iniconfig @ file:///home/lalo/.cache/pypoetry/artifacts/fa/b0/c6/10cfac68c9e6de9d2a1678366ca89fd9292b362c1760dbe758e41691cb/iniconfig-1.1.1-py2.py3-none-any.whl
isort @ file:///home/lalo/.cache/pypoetry/artifacts/13/0e/9d/0ac87b4f86576f57416f5d21432dec16c02955743e7afe51afe253a24b/isort-5.7.0-py3-none-any.whl
kiwisolver @ file:///home/lalo/.cache/pypoetry/artifacts/b0/3e/dd/9e102bc187a735e9c55414e8529e4eca4ccf7c94868e9eaf4e30612e20/kiwisolver-1.3.1-cp39-cp39-manylinux1_x86_64.whl
lazy-object-proxy @ file:///home/lalo/.cache/pypoetry/artifacts/18/04/70/fa7b9e82b3409e05c268ba4038442836717cc0255979a9e8cfff1a415f/lazy-object-proxy-1.4.3.tar.gz
matplotlib @ file:///home/lalo/.cache/pypoetry/artifacts/13/bf/94/d796b93a140aca4d596ceef9d6a6c8003b291f02bccaff1f140d77ae79/matplotlib-3.3.4-cp39-cp39-manylinux1_x86_64.whl
mccabe @ file:///home/lalo/.cache/pypoetry/artifacts/96/5e/5f/21ae5296697ca7f94de4da6e21d4936d74029c352a35202e4c339a4253/mccabe-0.6.1-py2.py3-none-any.whl
mypy-extensions @ file:///home/lalo/.cache/pypoetry/artifacts/92/45/bf/1807ce854ff668d92602207a37bfa9316def2a3f257bd03c4c5be4bc9b/mypy_extensions-0.4.3-py2.py3-none-any.whl
norfair==0.2.0
numpy @ file:///home/lalo/.cache/pypoetry/artifacts/43/46/5a/1c2baff5efec3c33813cc5751de96d64dfc46fd5b189f4e4a272227dc5/numpy-1.20.1-cp39-cp39-manylinux2010_x86_64.whl
packaging @ file:///home/lalo/.cache/pypoetry/artifacts/c2/76/aa/699ed723689e741411dfbdbc9b8c392079d54fb1e1a1b814df7be7cf47/packaging-20.9-py2.py3-none-any.whl
pathspec @ file:///home/lalo/.cache/pypoetry/artifacts/40/8b/b2/80a6945971d8475bcc04d09afec4845855dd74f68da6a4c18bbf8f7784/pathspec-0.8.1-py2.py3-none-any.whl
Pillow @ file:///home/lalo/.cache/pypoetry/artifacts/0a/a2/5b/2326e7164eba693d87968513704cd00843f2ff006e0d55a79c8e209f6a/Pillow-8.1.0-cp39-cp39-manylinux1_x86_64.whl
pluggy @ file:///home/lalo/.cache/pypoetry/artifacts/29/58/fc/ed8b7451d3ef91a6465024f5656141da996e7aafd4d41a1659629a75e7/pluggy-0.13.1-py2.py3-none-any.whl
py @ file:///home/lalo/.cache/pypoetry/artifacts/60/79/0b/c48bd9c2a989aa8b1eb7a67cd02b053c10734f2e4e5665f7995f09999c/py-1.10.0-py2.py3-none-any.whl
Pygments @ file:///home/lalo/.cache/pypoetry/artifacts/54/12/c1/dee5244df91f15f2bd40d20b21c37bb0067e7ec19e45606a530f902ce5/Pygments-2.7.4-py3-none-any.whl
pylint @ file:///home/lalo/.cache/pypoetry/artifacts/f5/28/9c/9c127841963caba0fa5310c92162143da8ad0b19de264fb03c7b25d79d/pylint-2.6.0-py3-none-any.whl
pyparsing @ file:///home/lalo/.cache/pypoetry/artifacts/92/0f/cf/effdcd5d76a6186df0969f85b3b030284ff8058936d5016540b5258ea3/pyparsing-2.4.7-py2.py3-none-any.whl
pytest @ file:///home/lalo/.cache/pypoetry/artifacts/61/a8/ec/1ad0d4114cf9e069c28315cff0a4774573e5b8656efa7a78d2f50c20e4/pytest-6.2.2-py3-none-any.whl
python-dateutil @ file:///home/lalo/.cache/pypoetry/artifacts/93/67/cf/49f56d9e954addcfc50e5ffc9faee013c2eb00c6d77d56c6a22cb33b54/python_dateutil-2.8.1-py2.py3-none-any.whl
regex @ file:///home/lalo/.cache/pypoetry/artifacts/a5/0c/c1/33400529698ff03037d95e34dd929002b124500316b658d69cd202001e/regex-2020.11.13-cp39-cp39-manylinux2014_x86_64.whl
rich @ file:///home/lalo/.cache/pypoetry/artifacts/f2/57/ad/10baedf4a5e5b07d7d209b917cb45ab685cc52e7dc5ee1df5b4a242e6f/rich-9.10.0-py3-none-any.whl
scipy @ file:///home/lalo/.cache/pypoetry/artifacts/80/05/33/eb27d3e0f20b100b6c5e76569aa3d6b0298e57258d270f782f6272a230/scipy-1.6.0-cp39-cp39-manylinux1_x86_64.whl
six @ file:///home/lalo/.cache/pypoetry/artifacts/dd/1c/65/ad0dea11136f5a869f072890a0eea955aa8fc35b90c85c55249fd3abfe/six-1.15.0-py2.py3-none-any.whl
toml @ file:///home/lalo/.cache/pypoetry/artifacts/6b/6a/c9/53b19f7870a77d855e8b05ecdc98193944e5d246dafe11bbcad850ecba/toml-0.10.2-py2.py3-none-any.whl
torch==1.8.1
tox @ file:///home/lalo/.cache/pypoetry/artifacts/4a/bf/25/63b41bd9fca4d7bf3f0c2fb33bc821783362f56633ca84a4df38e37036/tox-3.21.4-py2.py3-none-any.whl
typed-ast @ file:///home/lalo/.cache/pypoetry/artifacts/cd/79/52/46fe8f18205909c174875a57775caf1386309766aad7e887cc0d4b023d/typed_ast-1.4.2-cp39-cp39-manylinux1_x86_64.whl
typing-extensions==3.7.4.3
virtualenv @ file:///home/lalo/.cache/pypoetry/artifacts/c8/1f/8b/df062f7283a6f73a38f2d46224e85f15cfec69d9678701e06ddeb9c287/virtualenv-20.4.2-py2.py3-none-any.whl
wrapt==1.12.1
```
